Edit.java

```diff
@@ -1,33 +1,34 @@
 import java.util.ArrayList;
 
-public class Edit implements MenuItem<ArrayList<Contact>>{
+public class Edit implements Command{
 
     private final String name = "Edit";
     private final UI ui;
+    private final ArrayList<Contact> contacts;
 
-    public Edit(UI ui) {
+    public Edit(UI ui, ArrayList<Contact> contacts) {
         this.ui = ui;
+        this.contacts = contacts;
     }
 
     @Override
-    public String name() {
-        return name;
+    public void execute() {
+        Contact oldContact = userPickContact(contacts);
```

Comment:
Why not just get the index rather than the Contact?  Then it too could be inlined into 
contacts.set(contactToBeEdited(), updatedContact())